libs/@local/hash-graph-types/rust/src/ontology/data_type/lookup.rs

```diff
@@ -0,0 +1,46 @@
+use core::{borrow::Borrow, error::Error};
+
+use error_stack::{FutureExt as _, Report};
+use type_system::{
+    schema::{ClosedDataType, ConversionExpression, DataTypeReference, DataTypeUuid},
+    url::BaseUrl,
+};
+
+use crate::ontology::DataTypeWithMetadata;
+
+#[trait_variant::make(Send)]
+pub trait DataTypeLookup {
+    type DataTypeWithMetadata: Borrow<DataTypeWithMetadata> + Send;
```

Comment:
Are you using `Borrow` here for the additional guarantees, or would `AsRef` be sufficient here?

Reply:
`AsRef` would probably be sufficient. But as `Borrow` has stronger guarantees (at least semantically) I chose that. Do you think I should change it?

Reply:
Oh, and we use `Borrow` in other locations as well 😅

Reply:
I just wanted to make sure that you intended that those stronger guarantees are upheld^^